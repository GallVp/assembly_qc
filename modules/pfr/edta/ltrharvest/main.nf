process EDTA_LTRHARVEST {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' || workflow.containerEngine == 'apptainer' ?
        'https://depot.galaxyproject.org/singularity/edta:2.1.0--hdfd78af_1':
        'quay.io/biocontainers/edta:2.1.0--hdfd78af_1' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.gff3") , emit: gff3
    tuple val(meta), path("*.scn")  , emit: scn
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    /usr/local/share/EDTA/bin/LTR_HARVEST_parallel/LTR_HARVEST_parallel \\
        -seq $fasta \\
        $args \\
        -threads $task.cpus

    mv "${fasta}.harvest.combine.gff3" \\
        "${prefix}.gff3"

    mv "${fasta}.harvest.combine.scn" \\
        "${prefix}.scn"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        LTR_HARVEST_parallel: \$(/usr/local/share/EDTA/bin/LTR_HARVEST_parallel/LTR_HARVEST_parallel -h | sed -n '/Version/s/Version: //p')
        genometools: \$(gt --version | sed '1!d ; s/gt (GenomeTools) //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch "${prefix}.gff3"
    touch "${prefix}.scn"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        LTR_HARVEST_parallel: \$(/usr/local/share/EDTA/bin/LTR_HARVEST_parallel/LTR_HARVEST_parallel -h | sed -n '/Version/s/Version: //p')
        genometools: \$(gt --version | sed '1!d ; s/gt (GenomeTools) //')
    END_VERSIONS
    """
}
