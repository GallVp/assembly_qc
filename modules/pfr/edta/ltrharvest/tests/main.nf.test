nextflow_process {

    name "Test Process EDTA_LTRHARVEST"
    script "../main.nf"
    process "EDTA_LTRHARVEST"

    tag "modules"
    tag "modules_nfcore"
    tag "edta"
    tag "edta/ltrharvest"

    test("homo_sapiens-genome_21_fasta") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_21_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(path(process.out.versions[0]).text).match("script_versions") }
            )
        }

    }

    test("homo_sapiens-genome_fasta-stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(path(process.out.versions[0]).text).match("stub_versions") }
            )
        }

    }

}
