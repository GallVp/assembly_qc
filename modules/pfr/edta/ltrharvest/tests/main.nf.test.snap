{
    "homo_sapiens-genome_21_fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,da13c4ba22e44ef944ddec38aa72c468"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.scn:md5,65a6b80823a3f058142aed623a028a22"
                    ]
                ],
                "2": [
                    "versions.yml:md5,da7912721842adc402c15740be57751f"
                ],
                "gff3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,da13c4ba22e44ef944ddec38aa72c468"
                    ]
                ],
                "scn": [
                    [
                        {
                            "id": "test"
                        },
                        "test.scn:md5,65a6b80823a3f058142aed623a028a22"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,da7912721842adc402c15740be57751f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-02T21:08:07.746204"
    },
    "homo_sapiens-genome_fasta-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.scn:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,da7912721842adc402c15740be57751f"
                ],
                "gff3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff3:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "scn": [
                    [
                        {
                            "id": "test"
                        },
                        "test.scn:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,da7912721842adc402c15740be57751f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-02T21:08:13.587479"
    },
    "script_versions": {
        "content": [
            "\"EDTA_LTRHARVEST\":\n    LTR_HARVEST_parallel: v1.1\n    genometools: 1.6.2\n"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-02T21:08:07.763375"
    },
    "stub_versions": {
        "content": [
            "\"EDTA_LTRHARVEST\":\n    LTR_HARVEST_parallel: v1.1\n    genometools: 1.6.2\n"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-02T21:08:13.601529"
    }
}